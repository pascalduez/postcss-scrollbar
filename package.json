{
  "name": "postcss-scrollbar",
  "version": "0.4.0",
  "description": "PostCSS plugin enabling custom scrollbars",
  "keywords": [
    "css",
    "scrollbar",
    "postcss",
    "postcss-plugin"
  ],
  "author": {
    "name": "Pascal Duez",
    "url": "https://github.com/pascalduez"
  },
  "homepage": "https://github.com/pascalduez/postcss-scrollbar",
  "bugs": "https://github.com/pascalduez/postcss-scrollbar/issues",
  "repository": {
    "type": "git",
    "url": "git://github.com/pascalduez/postcss-scrollbar.git"
  },
  "license": "Unlicense",
  "files": [
    "dist",
    "CHANGELOG.md",
    "README.md",
    "UNLICENSE"
  ],
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "scripts": {
    "lint:js": "eslint src/ test/",
    "lint": "run-s lint:*",
    "format:check": "prettier -l '{src,test}/**/*.{js,css}'",
    "format:write": "prettier --write '{src,test}/**/*.{js,css}'",
    "validate": "run-s lint format:check",
    "test": "jest",
    "test:ci": "run-s 'test --coverage'",
    "develop": "babel-node develop/",
    "prebuild": "rm -rf dist/",
    "build": "rollup -c",
    "prepare": "run-s build",
    "prepublishOnly": "run-s validate test"
  },
  "dependencies": {
    "postcss": "^7.0.27",
    "postcss-selector-parser": "^6.0.9",
    "postcss-value-parser": "^4.2.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.17.6",
    "@babel/core": "^7.17.5",
    "@babel/eslint-parser": "^7.17.0",
    "@babel/node": "^7.16.8",
    "@babel/plugin-proposal-do-expressions": "^7.16.7",
    "@babel/preset-env": "^7.16.11",
    "babel-eslint": "^10.1.0",
    "common-tags": "^1.8.2",
    "eslint": "^8.9.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-config-prettier": "^8.4.0",
    "eslint-plugin-babel": "^5.3.1",
    "eslint-plugin-import": "^2.25.4",
    "eslint-plugin-jest": "^26.1.1",
    "jest": "^27.5.1",
    "npm-run-all": "^4.0.2",
    "postcss-reporter": "^6.0.1",
    "prettier": "^2.5.1",
    "rollup": "^2.68.0",
    "rollup-plugin-babel": "^4.4.0",
    "rollup-plugin-json": "^4.0.0"
  }
}
