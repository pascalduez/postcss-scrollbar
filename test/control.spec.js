import postcss from 'postcss';
import { stripIndent } from 'common-tags';
import { name } from '../package.json';
import plugin from '../src';

let from, to;
let input = stripIndent`
  .test {
    scrollbar-width: thin;
  }
`;

describe('control:', () => {
  test('no options', () =>
    postcss()
      .use(plugin)
      .process(input, { from, to })
      .then(result => {
        expect(result.css).toMatchSnapshot();
      }));

  test('with options', () =>
    postcss()
      .use(plugin({}))
      .process(input, { from, to })
      .then(result => {
        expect(result.css).toMatchSnapshot();
      }));

  test('PostCSS API', async () => {
    let processor = postcss();
    let result = await processor.use(plugin).process(input, { from, to });

    expect(result.css).toMatchSnapshot();
    expect(processor.plugins[0].postcssPlugin).toBe(name);
  });
});
