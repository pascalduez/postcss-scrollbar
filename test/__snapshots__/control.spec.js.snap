// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`control: PostCSS API 1`] = `
".test::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}
.test {
  -ms-overflow-style: auto;
  scrollbar-width: thin;
}"
`;

exports[`control: no options 1`] = `
".test::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}
.test {
  -ms-overflow-style: auto;
  scrollbar-width: thin;
}"
`;

exports[`control: with options 1`] = `
".test::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}
.test {
  -ms-overflow-style: auto;
  scrollbar-width: thin;
}"
`;
